### PR TITLE
Use validatejs in users forms

### PR DESCRIPTION
**Pull request by brunomichetti**
- Create a constraint file to validate each input
- validate function added in al the users forms
- remove unnecessary boolean validators
- standardize a way to clean the output

- TODO: validatejs in targets forms

### CODE REVIEW

File: src/components/users/EditProfileForm.js [hunk 1]
```diff
@@ -163,6 +182,7 @@ const EditProfileForm = ({ user, setEditProfile, newImg }) => {
 EditProfileForm.propTypes = {
   user: userShape,
   setEditProfile: func,
+  newImg: object,
```
Comment (tamarasuarezrod) [inline src/components/users/EditProfileForm.js:168]: is this an object? 🤔

File: src/components/users/ForgotPassword.js [hunk 2]
```diff
@@ -78,13 +85,15 @@ const ForgotPassword = ({ setForgotPassword }) => {
               inputName="email"
               inputValue={email}
               inputOnChange={handleChange}
-              error={isSubmitted && !email}
+              error={'email' in errors}
```
Comment (tamarasuarezrod) [inline src/components/users/ForgotPassword.js:89]: don't you have these names defined somewhere?

File: src/components/users/LoginForm.js [hunk 3]
```diff
@@ -17,31 +20,28 @@ const LoginForm = () => {
...
+    setInputs((inputs) => ({ ...inputs, [name]: value }));
   };
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    setIsSubmitted(true);
-    if (email && password) {
-      setCleanAlert(false);
+    var currentErrors = validate(inputs, loginConstraints) || {};
+    if (_.isEmpty(currentErrors)) {
```
Comment (tamarasuarezrod) [inline src/components/users/LoginForm.js]: I think you can use isEmpty from: `import { isEmpty } from 'lodash';` you don't need the underscore

**Review by tamarasuarezrod — APPROVED** [review]

File: src/components/users/EditProfileForm.js (hunk 1 above)
Comment (brunomichetti) [reply src/components/users/EditProfileForm.js:168]: I'm not quite sure the type of this object. Let me check

File: src/components/users/ForgotPassword.js (hunk 2 above)
Comment (brunomichetti) [reply src/components/users/ForgotPassword.js:89]: What do you mean?

File: src/components/users/LoginForm.js [hunk 4]
```diff
@@ -17,31 +20,28 @@ const LoginForm = () => {
...
+    setErrors(omit(errors, name));
+    setInputs((inputs) => ({ ...inputs, [name]: value }));
   };
 
   const handleSubmit = (e) => {
     e.preventDefault();
-    setIsSubmitted(true);
-    if (email && password) {
-      setCleanAlert(false);
+    var currentErrors = validate(inputs, loginConstraints) || {};
```
Comment (manema) [inline src/components/users/LoginForm.js]: any reason to use **var** variables? I would prefer use ECMA5 variables(const, let) as long as possible.

File: src/helpers/constraints.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
+export var validate = require('validate.js');
```
Comment (manema) [inline src/helpers/constraints.js]: are you using babel or any other compiler? if so try _import validate from 'validate.js';_ instead. Besides the constraints below should be const.

File: src/components/users/EditProfileForm.js (hunk 1 above)
Comment (manema) [reply src/components/users/EditProfileForm.js:168]: Anyway, to use object type itself in propTypes is pretty useless. The idea behind to use propTypes is that you got warned each time a component prop is not what it's supposed to be. If you use object per se, whatever object this component gets will be ok, if I pass down { string: 'hola' } would be ok, same for { a: {b: 3 }}. Make sure to give an accurate shape to the propTypes so you will be really taking advantage of them. :slightly_smiling_face:

Comment (brunomichetti) [reply src/components/users/EditProfileForm.js:168]: Yes is an object. A File object

**Review by manema — APPROVED** [review]
Far better 👏 👏 🚀

**Pull request merged**